tools/fds/server.cpp

```diff
@@ -228,10 +227,20 @@ int main (
 
         // Print running server attributes
         cout << "### Server is running ###" << endl;
+        cout << "  Participant Type:   " << rtps.builtin.discovery_config.discoveryProtocol << endl;
         cout << "  Server ID:          " << server_id << endl;
         cout << "  Server GUID prefix: " << pServer->getGuid().guidPrefix << endl;
-        cout << "  Server Address:     " << locator << endl;
-        cout << "  Participant Type    " << rtps.builtin.discovery_config.discoveryProtocol << endl;
+        cout << "  Server Addresses:   ";
+        for (auto locator_it = att->rtps.builtin.metatrafficUnicastLocatorList.begin();
+                locator_it != att->rtps.builtin.metatrafficUnicastLocatorList.end();)
+        {
+            cout << *locator_it;
+            if (++locator_it != att->rtps.builtin.metatrafficUnicastLocatorList.end())
+            {
+                cout << std::endl << "                      ";
+            }
```

Comment:
```suggestion
        for (auto locator : att->rtps.builtin.metatrafficUnicastLocatorList)
        {
            cout << locator << std::endl << "                      ";
        }
```

I suggest using the ranged for. Also, LocatorList class have operator `<<` overloaded so we can also take advantage of this.

Reply:
I actually was doing that in the beginning and then at some point switched to the iterators to see if that solved another problem I was having. I'll go back to range for 👌🏼

Reply:
After internal discussion, we have decided to leave the iterator to be able to do different things for the last element